test/locking.js

```diff
@@ -44,196 +42,291 @@ contract('Staking app, Locking', ([owner, user1, user2]) => {
+    await approveStakeAndLock(user1)
+
+    await assertRevert(staking.increaseLockAllowance(user1, 0))
+  })
+
+  it('decreases allowance of existing lock, owner', async () => {
+    await approveAndStake()
+    await staking.allowManagerAndLock(DEFAULT_LOCK_AMOUNT, user1, DEFAULT_LOCK_AMOUNT + 1, EMPTY_DATA)
+
+    await staking.decreaseLockAllowance(owner, user1, 1, { from: owner })
+
+    const [ , _allowance  ] = await staking.getLock(owner, user1)
+    assert.equal(_allowance, DEFAULT_LOCK_AMOUNT, "allowed amount should match")
+  })
+
+  it('decreases allowance of existing lock, manager', async () => {
+    await approveAndStake()
+    await staking.allowManagerAndLock(DEFAULT_LOCK_AMOUNT, user1, DEFAULT_LOCK_AMOUNT + 1, EMPTY_DATA)
+
+    await staking.decreaseLockAllowance(owner, user1, 1, { from: user1 })
+
+    const [ , _allowance  ] = await staking.getLock(owner, user1)
+    assert.equal(_allowance, DEFAULT_LOCK_AMOUNT, "allowed amount should match")
+  })
+
+  it('fails decreasing allowance of existing lock by 0', async () => {
+    await approveStakeAndLock(user1)
+
+    await assertRevert(staking.decreaseLockAllowance(owner, user1, 0))
+  })
+
+  it('fails decreasing allowance of existing lock to 0', async () => {
+    await approveStakeAndLock(user1)
+
+    await staking.decreaseLockAmount(owner, user1, DEFAULT_LOCK_AMOUNT, { from: user1 })
+
+    await assertRevert(staking.decreaseLockAllowance(owner, user1, DEFAULT_LOCK_AMOUNT))
+  })
+
+  it('fails decreasing allowance to less than lock', async () => {
+    await approveAndStake()
+    await staking.allowManagerAndLock(DEFAULT_LOCK_AMOUNT, user1, DEFAULT_LOCK_AMOUNT + 1, EMPTY_DATA)
+
+    await assertRevert(staking.decreaseLockAllowance(owner, user1, 2))
+  })
+
+  it('fails decreasing allowance by 3rd party', async () => {
+    await approveAndStake()
+    await staking.allowManagerAndLock(DEFAULT_LOCK_AMOUNT, user1, DEFAULT_LOCK_AMOUNT + 1, EMPTY_DATA)
+
+    await assertRevert(staking.decreaseLockAllowance(owner, user1, 1, { from: user2 }))
+  })
+
+  it('increases amount of existing lock', async () => {
+    await approveStakeAndLock(user1)
+
+    await approveAndStake()
+    await staking.increaseLockAllowance(user1, DEFAULT_LOCK_AMOUNT)
+    await staking.increaseLockAmount(owner, user1, DEFAULT_LOCK_AMOUNT)
+
+    const [ _amount,  ] = await staking.getLock(owner, user1)
+    assert.equal(_amount, 2 * DEFAULT_LOCK_AMOUNT, "locked amount should match")
+  })
+
+  it('fails increasing lock with 0 tokens', async () => {
+    await approveStakeAndLock(user1)
+
+    await approveAndStake()
+    await assertRevert(staking.increaseLockAmount(owner, user1, 0))
+  })
+
+  it('fails increasing lock with more tokens than staked', async () => {
+    await approveStakeAndLock(user1)
+
+    await approveAndStake()
+    await assertRevert(staking.increaseLockAmount(owner, user1, 2 * DEFAULT_STAKE_AMOUNT + 1))
+  })
+
+  it('unlocks with only 1 lock, EOA manager', async () => {
```

Comment:
we could group these scenarios in different blocks of tests by types of manager

Reply:
I will do that here: https://github.com/aragon/staking/issues/17